token/js/src/instructions/associatedTokenAccount.ts

```diff
@@ -64,6 +65,31 @@ export function createAssociatedTokenAccountIdempotentInstruction(
     );
 }
 
+/**
+ * Construct a createAssociatedTokenAccountIdempotentInstructionShort instruction
+ *
+ * @param payer                    Payer of the initialization fees
+ * @param owner                    Owner of the new account
+ * @param mint                     Token mint account
+ * @param allowOwnerOffCurve       Allow the owner account to be a PDA (Program Derived Address)
+ * @param programId                SPL Token program account
+ * @param associatedTokenProgramId SPL Associated Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createAssociatedTokenAccountIdempotentInstructionShort(
+    payer: PublicKey,
+    owner: PublicKey,
+    mint: PublicKey,
+    allowOwnerOffCurve = true,
+    programId = TOKEN_PROGRAM_ID,
+    associatedTokenProgramId = ASSOCIATED_TOKEN_PROGRAM_ID,
+) {
+    const associatedToken = getAssociatedTokenAddressSync(mint, owner, allowOwnerOffCurve);
+
+    return createAssociatedTokenAccountIdempotentInstruction(payer, owner, mint, programId, associatedTokenProgramId);
```

Comment:
This isn't using the associated token account address that was just derived, so it's incorrect. You'll need to do:
```suggestion
    return createAssociatedTokenAccountIdempotentInstruction(payer, associatedToken, owner, mint, programId, associatedTokenProgramId);
```